upgrade/v2.5/README.md

```diff
@@ -1,52 +1,47 @@
-# Calico v2.5.0 Upgrade Procedure
+# Migrating Kubernetes datastore data before upgrading to Calico v2.5.0 or later
```

Comment:
This doc is for both migration and updating calico version, not just for migration. I think we can just say `Upgrading Calico to v2.5.0+ with Kubernetes datastore backend` or something similar? WDYT?